core/src/cluster_info.rs

```diff
@@ -1756,11 +1764,16 @@ impl ClusterInfo {
                             .unwrap()
                             .set_shred_version(entrypoint.shred_version);
                         self.insert_self();
-                        *self.entrypoint.write().unwrap() = Some(entrypoint);
-                        *adopt_shred_version = false;
                     }
                 }
+
+                // Update the entrypoint's id so future entrypoint pulls correctly reference it
+                *self.entrypoint.write().unwrap() = Some(entrypoint);
+                *entrypoint_processed = true;
```

Comment:
Wouldn't this prevent `shred_version` to ever get updated, if on the first pass `entrypoint.shred_version == 0`?

Reply:
I couldn't think of many good reasons why entrypoint.shred_version could ever be 0 on the first pass but not the second pass.   Only reason I can think of is if the entrypoint itself is restarted.

Reply:
I guess that's reason enough 👍🏼